src/Jwt/Udb3Token.php

```diff
@@ -45,7 +50,28 @@ public function isAccessToken(): bool
         // This does not 100% guarantee that the token is an access token, because an access token does not have an azp
         // if it has no specific aud. However we require our integrators to always include the "https://api.publiq.be"
         // aud, so access tokens should always have an azp in our case.
-        return !is_null($this->getClientId());
+        if ($this->getClientId() === null) {
+            return false;
+        }
+
+        if (!$this->token->hasClaim('sub')) {
+            return true;
```

Comment:
@bertramakers I wonder if we should not beter return false here? Because a `sub` is a requirement for both an access and client token?

Reply:
I'm not sure what this change is meant for. A client access token is technically an access token, just like a user access token. Are you trying to differentiate between user and client access tokens?

I would not change this specific method then, because we need to know it's a an access token (regardless of if it's a client or user access token) to check the EntryAPI permissions here https://github.com/cultuurnet/udb3-silex/blob/b34a30e4a7faa1dfef32ff93f90924dc561905cd/src/Jwt/JwtV2Validator.php#L38

Reply:
The different kind of Auth0 tokens are as follows:

- Access tokens
  - Client access tokens
  - User access tokens
- Id tokens